backend/api/Areas/Tools/Helpers/ImportPropertiesHelper.cs

```diff
@@ -0,0 +1,410 @@
+                        break;
+                    case "MTICS":
+                        property.Agency = "CITZ";
+                        break;
+                }
+
+                // Fix postal.
+                property.Postal = new string(property.Postal?.Replace(" ", "").Take(6).ToArray());
+
+                var agency = GetOrCreateAgency(property);
+
+                if (String.Compare(property.PropertyType, "Land") == 0)
+                {
+                    entities.Add(AddUpdateParcel(property, pid, agency));
+                }
+                else if (String.Compare(property.PropertyType, "Building") == 0)
+                {
+                    var parcel = AddUpdateBuilding(property, pid, agency);
+                    if (!entities.Any(p => p.ParcelId == parcel.ParcelId))
+                    {
+                        entities.Add(parcel);
+                    }
+                }
+            }
+
+            return entities;
+        }
+
+        /// <summary>
+        /// Get or create a new agency for the specified property.
+        /// </summary>
+        /// <param name="property"></param>
+        /// <param name="agencies"></param>
+        /// <returns></returns>
+        private Entity.Agency GetOrCreateAgency(PropertyModel property)
+        {
+            // Find the agency or create a new one.
+            var agency = _agencies.FirstOrDefault(a => a.Code == property.Agency.Trim());
+
+            if (agency == null)
+            {
+                agency = new Entity.Agency(property.Agency.Trim(), property.Agency);
+                _pimsAdminService.Agency.Add(agency);
+                _agencies.Add(agency);
+                _logger.LogDebug($"Adding agency '{agency.Code}' - '{agency.Name}'.");
+            }
+
+            if (!String.IsNullOrWhiteSpace(property.SubAgency))
+            {
+                var code = new string(property.SubAgency.GetFirstLetterOfEachWord(true).Take(6).ToArray());
+                var sub_agency = _agencies.FirstOrDefault(a => a.Code == code.Trim() && a.ParentId == agency.Id);
+
+                if (sub_agency == null)
+                {
+                    sub_agency = new Entity.Agency(code.Trim(), property.SubAgency)
+                    {
+                    ParentId = agency.Id,
+                    Parent = agency
+                    };
+                    _agencies.Add(sub_agency);
+                    _logger.LogDebug($"Adding sub-agency '{sub_agency.Code}' - '{agency.Name}', parent: '{sub_agency.Parent.Code}'.");
+                }
+
+                return sub_agency;
+            }
+
+            return agency;
+        }
+
+        /// <summary>
+        /// Get or create a new city for the specified code.
+        /// This is a recursive function so that it can generate a new city with a unique code.
+        /// It has a limitation of which the code must be 4 characters and therefore it can only handle 9 variations (i.e. BRe1, BRe2...)
```

Comment:
if only 9 variations are allowed, shouldn't we put a cap on the max size of attempts?

Reply:
The result of going over would be a DbUpdateException as they would attempt to insert a city with a code with a length greater than 4 characters.  I'm hesitant to limit the use here, as it would indicate we probably need to modify the DB to accept more characters.